documentation/intermediate-ca.md

```diff
@@ -0,0 +1,191 @@
+# Intermediate CA support with Puppet Server
+
+## Configuration guide
+
+When Puppet Server or PE is installed it will autogenerate a number of certificates. Regenerating all of these certificates can be a time consuming process, but this problem can be side stepped by generating an intermediate CA certificate that reuses the existing key pair that Puppet Server/PE has already created.
+
+### Configure the Puppet agent `certificate_revocation` checking
+
+Intermediate CA support is incompatible with full chain revocation checking, as Puppet is unable to load multiple CRLs. When an intermediate CA certificate is in use Puppet must have the `certificate_revocation` setting either set to `false`, or `leaf`. By default Puppet will attempt to perform full chain checking and will fail when encountering a non-root CA certificate, so this setting **must** be reconfigured on all Puppet agents before continuing.
+
+    puppet config set --section main certificate_revocation leaf
+
+See [PUP-3788](https://tickets.puppetlabs.com/browse/PUP-3788) for more information on chain revocation checking.
```

Comment:
I wonder if we should link to a docs site page or something rather than a bug..

Reply:
I'm amenable to that, but I'd prefer a documentarian to make a decision on that. Unfortunately as you'll see later in the document there are a lot of bugs/sharp edges here so I'm not sure how to approach documentation for so many issues.